internal/topology/variables/patch_validation.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package variables
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"text/template"
+
+	"github.com/dolmen-go/jsonptr"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// ValidatePatches returns errors if the Patches in the ClusterClass violate any validation rules.
+func ValidatePatches(patches []clusterv1.ClusterClassPatch, variables []clusterv1.ClusterClassVariable, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	names := sets.String{}
+	for i, patch := range patches {
+		allErrs = append(
+			allErrs,
+			validatePatch(patch, names, variables, path.Index(i))...,
+		)
+		names.Insert(patch.Name)
+	}
+	return allErrs
+}
+
+func validatePatch(patch clusterv1.ClusterClassPatch, names sets.String, variables []clusterv1.ClusterClassVariable, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs,
+		validatePatchName(patch, names, path)...,
+	)
```

Comment:
```suggestion
	allErrs = append(allErrs,validatePatchName(patch, names, path)...)
```
nit. Maybe just inline it. To me it reads more fluently that way.

(also in other cases)